lib/elixir/lib/calendar/naive_datetime.ex

```diff
@@ -317,7 +317,7 @@ defmodule NaiveDateTime do
           Calendar.hour(),
           Calendar.minute(),
           Calendar.second(),
-          Calendar.microsecond() | non_neg_integer,
```

Comment:
I'm not sure here, but the rest of the file uses `non_neg_integer()` spec notation, so I tweaked the spec here.